### ISSUE TITLE
[Issue] The TagRouter maybe have one issue

### ISSUE THREAD

**Issue by Hinsteny** [apache/dubbo#5971]
- [ X] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [ X] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 2.7.7
* Operating System version: macOS
* Java version: 1.8

### Steps to reproduce this issue

1. Two provider service with same interface [GreetingsService] and different service-port and provider-tag[dubbo-tag=grey];--- Provider-A(no tag), Provider-B(with tag), 
2. Start one Consumer to call the GreetingsService with no tag; --- Consumer-X(no tag), 

### Expected Result

What do you expected from the above steps？

> I think and need the consumer to call the provider with no tag grey;

### Actual Result

What actually happens?

> The consumer always call the provider with tag grey;

### Reason 
When dubbo initial the consumer service reference, it will merge the all provider Parameters to the consumer Parameter map; then do the provider select will use the provider's tag to select provider;

TagRouter
```
   @Override
    public <T> List<Invoker<T>> route(List<Invoker<T>> invokers, URL url, Invocation invocation) throws RpcException {
        if (CollectionUtils.isEmpty(invokers)) {
            return invokers;
        }

        // since the rule can be changed by config center, we should copy one to use.
        final TagRouterRule tagRouterRuleCopy = tagRouterRule;
        if (tagRouterRuleCopy == null || !tagRouterRuleCopy.isValid() || !tagRouterRuleCopy.isEnabled()) {
            return filterUsingStaticTag(invokers, url, invocation);
        }

        List<Invoker<T>> result = invokers;
        String tag = StringUtils.isEmpty(invocation.getAttachment(TAG_KEY)) ? url.getParameter(TAG_KEY) :
                invocation.getAttachment(TAG_KEY);

        // if we are requesting for a Provider with a specific tag
        if (StringUtils.isNotEmpty(tag)) {
```

**Comment by zhouhaopeng** [comment]
please offer your router config

**Comment by Hinsteny** [comment]
## Prvider
1. Provider-A: no tag
2. Provider-B: with tag dubbo.tag=grey

## Consumer
1. Consumer-X: no tag, do router select get B(should get A);
2. Consumer-Y: with tag dubbo.tag=grey, do router select get A;
3. Consumer-Y: with tag dubbo.tag=grey_, do router select get A;

**Comment by Hinsteny** [comment]
I have made a PR refer [5972](https://github.com/apache/dubbo/pull/5972)

**Comment by cvictory** [comment]
最好能提供一个可运行的demo来验证。

**Comment by cvictory** [comment]
可以贴一个github的demo

**Comment by eightroad** [comment]
any update?
I have the same problem，seems when consumers do not set dubbo.consumer.tag, it will be overridden by provider config.
absolutely, the "dubbo.tag" not exists at this time. 

org.apache.dubbo.registry.integration.RegistryDirectory#mergeUrl
        this.overrideDirectoryUrl = this.overrideDirectoryUrl.addParametersIfAbsent(providerUrl.getParameters()); // Merge the provider side parameters

**Comment by moonlight035** [comment]
`this.overrideDirectoryUrl = this.overrideDirectoryUrl.addParametersIfAbsent(providerUrl.getParameters());`
当consumer没有配置tag时，会默认将provider的tag保存过来，导致consumer有了一个默认tag，我觉得这里不应该保存provider的tag

**Comment by 24kpure** [comment]
i suffer from this problem  for these code .Maybe some provider config params are not suitable to override in consumer,for instance, dubbo.tag.

`this.overrideDirectoryUrl = this.overrideDirectoryUrl.addParametersIfAbsent(providerUrl.getParameters());`

**Comment by KevinClair** [comment]
Hello,is it has been solved in dubbo version `2.7.7`?i get the same problem.

**Comment by plusmancn** [comment]
Duplicate of #6426, and the bug has been fixed in 2.7.9

**Comment by plusmancn** [comment]
@AlbumenJ This issue can be closed.
